File: app/src/main/java/br/com/soulblighter/lodjinha/model/Categoria.java
package br.com.soulblighter.lodjinha.model;

import android.os.Parcel;
import android.os.Parcelable;

public class Categoria implements Parcelable {

    private int id;
    private String descricao;
    private String urlImagem;

    public Categoria() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public String getUrlImagem() {
        return urlImagem;
    }

    public void setUrlImagem(String urlImagem) {
        this.urlImagem = urlImagem;
    }

    public final static Parcelable.Creator<Categoria> CREATOR = new Creator<Categoria>() {
        public Categoria createFromParcel(Parcel in) {
            return new Categoria(in);
        }

        public Categoria[] newArray(int size) {
            return (new Categoria[size]);
        }
    };

    protected Categoria(Parcel in) {
        this.id = ((int) in.readValue((int.class.getClassLoader())));
        this.descricao = ((String) in.readValue((String.class.getClassLoader())));
        this.urlImagem = ((String) in.readValue((String.class.getClassLoader())));
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(id);
        dest.writeValue(descricao);
        dest.writeValue(urlImagem);
    }

    public int describeContents() {
        return 0;
    }
}

File: app/src/main/java/br/com/amedigital/challenge_android/models/entity/Categoria.kt
package br.com.amedigital.challenge_android.models.entity

import android.os.Parcel
import android.os.Parcelable
import androidx.room.Entity

@Suppress("NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS", "unused")
@Entity(primaryKeys = [("id")])
data class Categoria (
    val id: Int,
    val descricao: String,
    val urlImagem: String
) : Parcelable {
    constructor(source: Parcel) : this(
        source.readInt(),
        source.readString(),
        source.readString()
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) = with(dest) {
        writeInt(id)
        writeString(descricao)
        writeString(urlImagem)
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<Categoria> = object : Parcelable.Creator<Categoria> {
            override fun createFromParcel(source: Parcel): Categoria = Categoria(source)
            override fun newArray(size: Int): Array<Categoria?> = arrayOfNulls(size)
        }
    }
}